src/circuit_proof/assignment.rs

```diff
@@ -137,3 +138,32 @@ impl Try for Assignment {
         Assignment::Value(v)
     }
 }
+
+impl ConditionallySelectable for Assignment {
+    // This function should execute in constant time for a and b of the same type.
+    // So, if a and b are both of type Assignment::Value(), then the comparison will
+    // execute in constant time regardless of their value assignments.
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        match (a, b) {
+            (Assignment::Value(a_val), Assignment::Value(b_val)) => {
+                let mut out_val = a_val.clone();
+                out_val.conditional_assign(&b_val, choice);
+                Assignment::from(out_val)
```

Comment:
This could have a FIXME note with a reference to the upstream issue

Reply:
done!